packages/core/src/scores/run-experiment.ts

```diff
@@ -0,0 +1,167 @@
+    });
+  }
+
+  if (scorers.length === 0) {
+    throw new MastraError({
+      domain: 'SCORER',
+      id: 'RUN_EXPERIMENT_FAILED_NO_SCORERS_PROVIDED',
+      category: 'USER',
+      text: 'Failed to run experiment: No scorers provided',
+    });
+  }
+
+  if (!target) {
+    throw new MastraError({
+      domain: 'SCORER',
+      id: 'RUN_EXPERIMENT_FAILED_NO_TARGET_PROVIDED',
+      category: 'USER',
+      text: 'Failed to run experiment: No target provided',
+    });
+  }
+
+  await pMap(
+    data,
+    async item => {
+      let targetResult: any;
+      try {
+        targetResult = await target.generate(item.input, {
+          scorers: {},
+          returnScorerData: true,
+          runtimeContext: item.runtimeContext,
+        });
+      } catch (error) {
+        throw new MastraError(
+          {
+            domain: 'SCORER',
+            id: 'RUN_EXPERIMENT_TARGET_FAILED_TO_GENERATE_RESULT',
+            category: 'USER',
+            text: 'Failed to run experiment: Error generating result from target',
+            details: {
+              item: JSON.stringify(item),
+            },
+          },
+          error,
+        );
+      }
+
+      const scorerResults: ScorerResults<TScorer> = {} as ScorerResults<TScorer>;
+      for (const scorer of scorers) {
+        try {
+          const score = await scorer.run({
+            input: targetResult.scoringData?.input,
+            output: targetResult.scoringData?.output,
+            groundTruth: item.groundTruth,
+            runtimeContext: item.runtimeContext,
+          });
+
+          scorerResults[scorer.name as keyof ScorerResults<TScorer>] =
+            score as ScorerResults<TScorer>[typeof scorer.name];
+        } catch (error) {
+          throw new MastraError(
+            {
+              domain: 'SCORER',
+              id: 'RUN_EXPERIMENT_SCORER_FAILED_TO_SCORE_RESULT',
+              category: 'USER',
+              text: `Failed to run experiment: Error running scorer ${scorer.name}`,
+              details: {
+                scorerName: scorer.name,
+                item: JSON.stringify(item),
+              },
+            },
+            error,
+          );
+        }
+      }
+
+      for (const [scorerName, result] of Object.entries(scorerResults)) {
+        if (!scoreAccumulators[scorerName]) {
+          scoreAccumulators[scorerName] = [];
+        }
+        scoreAccumulators[scorerName].push((result as { score: number }).score);
+      }
+
+      if (onItemComplete) {
+        onItemComplete({ item, targetResult, scorerResults });
+      }
+
+      totalItems++;
+    },
+    { concurrency },
+  );
+
+  const averageScores: Record<string, number> = {};
+  for (const [scorerName, scores] of Object.entries(scoreAccumulators)) {
+    averageScores[scorerName] = scores.reduce((a, b) => a + b, 0) / scores.length;
+  }
+
+  return {
+    scores: averageScores,
+    summary: {
+      totalItems,
+      duration: Date.now() - startTime,
```

Comment:
ah, i guess it is quite easy currently.. but if in the future, scores run asyncronously, is this still easy to provide?